### ISSUE TITLE
Don't use --show-web-server-device for newer Flutter SDKs

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#4409]
If https://github.com/flutter/flutter/pull/121373 remains, we need to stop trying to send this flag for newer SDK versions.

**Comment by DanTup** [comment]
@christopherfujino what's your plan for https://github.com/flutter/flutter/pull/121373? I think in order to avoid breaking anything, we need to:

1. change the default in Flutter, but keep the flag (so `--show-web-server-device` does nothing)
2. use the relevant version number to allow Dart-Code to skip sending the flag to new SDKs (but keep sending it to old)
3. remove the flag from Flutter some time after (2) has shipped

I can't make any change here until I know a version number where the flag is no longer necessary (but still accepted).

**Comment by christopherfujino** [comment]
> @christopherfujino what's your plan for [flutter/flutter#121373](https://github.com/flutter/flutter/pull/121373)? I think in order to avoid breaking anything, we need to:
> 
> 1. change the default in Flutter, but keep the flag (so `--show-web-server-device` does nothing)
> 2. use the relevant version number to allow Dart-Code to skip sending the flag to new SDKs (but keep sending it to old)
> 3. remove the flag from Flutter some time after (2) has shipped
> 
> I can't make any change here until I know a version number where the flag is no longer necessary (but still accepted).

That makes sense. @eliasyishak can you coordinate with @DanTup to re-land this without breaking Dart-Code?

**Comment by eliasyishak** [comment]
Just seeing this now while cleaning up some issues/PRs on my end; is this still something we need to address? Apologies on the late response, I think I was OOO this week.

If I can help still, let me know! @DanTup

**Comment by DanTup** [comment]
@eliasyishak if the plan is to re-land https://github.com/flutter/flutter/pull/121373, then it needs to be done in a non-breaking way. Then we can update Dart-Code here to stop using it when it's not required, before eventually removing it completely from Flutter.

I don't know what the motivation is for making that change though (I believe the flag was originally added to avoid confusion from users picking `web-server` over `chrome` which wasn't optimal since it requires the Dart debug extension).

**Comment by eliasyishak** [comment]
Ah yep, I just saw the revert happening in https://github.com/flutter/flutter/pull/121532, @christopherfujino is this something we still want to remove?

**Comment by christopherfujino** [comment]
> Ah yep, I just saw the revert happening in [flutter/flutter#121532](https://github.com/flutter/flutter/pull/121532), @christopherfujino is this something we still want to remove?

Yes, the reverted PR was solving: https://github.com/flutter/flutter/issues/120691

**Comment by DanTup** [comment]
Closing this because it's not currently actionable until it's clear what the plan/goal is for the Flutter change. We need to land non-breaking changes in Flutter first before we can change how Dart-Code sends this flag. If a Flutter change is to go ahead, we can re-open this.
